hack/update-staging-godeps.sh

```diff
@@ -71,7 +71,9 @@ function updateGodepManifest() {
     go run "${KUBE_ROOT}/staging/godeps-json-updater.go" --godeps-file="${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json" --client-go-import-path="k8s.io/${repo}"
 
     # commit so that following repos do not see this repo as dirty
-    git add vendor >/dev/null
+    # pipe through true so that command script won't error if there are no
+    # files to add
+    git add vendor/ >/dev/null || true
```

Comment:
Okay! So I think I have an answer here, and thank you for your patience, @sttts!

I started with a clean environment and gopath, and tinkered around with things. It seems that godep v74 has an issue when running in a go1.8 environment. I'm not sure what it is specifically, but using godep v79 the script works like a charm under either go1.7.5 or go1.8.

Is there a reason we are stuck on v74? If not, I'll close this PR, and open up a different one to bump the scripts to use v79.